pkg/controller/node/status.go

```diff
@@ -323,21 +333,29 @@ func isNodeMCDFailing(node *corev1.Node) bool {
 // getUpdatedMachines filters the provided nodes to return the nodes whose
 // current config matches the desired config, which also matches the target config,
 // and the "done" flag is set.
-func getUpdatedMachines(pool *mcfgv1.MachineConfigPool, nodes []*corev1.Node) []*corev1.Node {
+func getUpdatedMachines(pool *mcfgv1.MachineConfigPool, nodes []*corev1.Node, mosc *mcfgv1alpha1.MachineOSConfig, mosb *mcfgv1alpha1.MachineOSBuild, layered bool) []*corev1.Node {
 	var updated []*corev1.Node
 	for _, node := range nodes {
 		lns := ctrlcommon.NewLayeredNodeState(node)
-		if lns.IsDoneAt(pool) {
-			updated = append(updated, node)
+		if mosb != nil && mosc != nil {
+			mosbState := ctrlcommon.NewMachineOSBuildState(mosb)
+			if layered && mosbState.IsBuildSuccess() && mosb.Spec.DesiredConfig.Name == pool.Status.Configuration.Name {
```

Comment:
this should be looking for node annotation currentimage==desiredImage, otherwise this will never be true as nothing can set it (this is where the status is set)